app/src/main/java/com/seoulfitu/android/ui/main/MainActivity.kt

```diff
@@ -96,23 +105,44 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
-    private fun checkLocationPermission() {
-        if (checkSelfPermission(this, ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
-            locationPermissionRequest.launch(locationPermissions)
-            return
-        }
+    private fun requestLocationPermission() {
+        locationPermissionRequest.launch(ACCESS_FINE_LOCATION)
     }
 
     private fun setForecast() {
-        checkLocationPermission()
+        if (!isLocationPermissionGranted) {
+            requestLocationPermission()
+            return
+        }
         val fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
-        if (ContextCompat.checkSelfPermission(this, ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
-            fusedLocationClient.lastLocation
-                .addOnSuccessListener { location: Location? ->
-                    location?.let { viewModel.fetchWeather(it.latitude, location.longitude) }
-                    location?.let { viewModel.fetchParticulateMatter(it.latitude, location.longitude) }
-                }
+        if (ContextCompat.checkSelfPermission(this, ACCESS_FINE_LOCATION) != PERMISSION_DENIED) {
+            isLocationPermissionGranted = true
+            fusedLocationClient.getCurrentLocation(
+                createCurrentLocationRequest(),
+                createCancellationToken()
+            ).addOnSuccessListener { location: Location? ->
+                location?.let { viewModel.fetchWeather(it.latitude, it.longitude) }
+                location?.let { viewModel.fetchParticulateMatter(it.latitude, it.longitude) }
```

Comment:
```suggestion
 location?.let {
                    viewModel.fetchWeather(it.latitude, it.longitude)
                    viewModel.fetchParticulateMatter(it.latitude, it.longitude)
                }
```
엄청 사소한건데 이렇게 적어도 될 것 같아용 히히히

Reply:
오 더 깔끔하네요!! 이렇게 수정했습니다!